File: python/3_prime_palindrome.py
import math

def is_palindrome(n):
    s = str(n)

    return s == s[::-1]

def is_prime(n):
    if n <= 1:
        return False

    if n % 2 == 0 and n != 2:
        return False

    if n == 2:
        return True

    root = math.sqrt(n)
    i    = 3
    while i <= root:
        if n % i == 0:
            return False

        i += 2

    return True

i = 999
while i > 0:
    if not is_palindrome(i):
        i -= 1
        continue

    if not is_prime(i):
        i -= 1
        continue

    print i
    break

File: javascript/3_prime_palindrome.js
function is_palindrome(n) {
	var s = n.toString();
	var length = s.length;

	for (var i = 0; i <= length / 2; i++) {
		if (s[i] != s[length - i - 1])
			return false;
	}

	return true;
}

function is_prime(n) {
	if (n <= 1) {
		return false;
	}
	else if (n == 2) {
		return true;
	}
	else if (n % 2 === 0) {
		return false;
	}

	var root = Math.sqrt(n);
	for (var i = 3; i <= root; i++) {
		if (n % i === 0) {
			return false;
		}
	}

	return true;
}

for (var i = 999; i >= 3; i -= 2) {
	if (!is_palindrome(i) || !is_prime(i)) {
		continue;
	}

	console.log(i);
	break;
}

